Classes/KIFUITestActor.m

```diff
@@ -622,22 +622,38 @@ - (void)selectPickerValue:(NSArray *)pickerColumnValues pickerType:(KIFPickerTyp
         }
         // Find the picker view
         UIPickerView *pickerView = nil;
+        NSInteger firstRowToSearchIndex = 0;
         switch (pickerType)
         {
             case KIFUIDatePicker:
+            {
                 pickerView = [[[[UIApplication sharedApplication] datePickerWindow] subviewsWithClassNameOrSuperClassNamePrefix:@"UIPickerView"] lastObject];
                 KIFTestCondition(pickerView, error, @"No picker view is present");
+                
+                // If minimum date limit exist, search the value from minimum and beyond.
+                UIDatePicker *datePicker = (UIDatePicker *)pickerView;
+                if (datePicker.datePickerMode == UIDatePickerModeDateAndTime
+                    && datePicker.minimumDate != nil) {
+                    
+                    [datePicker setDate:datePicker.minimumDate];
```

Comment:
Ok, so I took a closer look and now understand the problem a bit better. The problem with `UIDatePickerModeDateAndTime` is that it has multiple years worth of values matching `Feb 28`, but it isn't explicitly called out anywhere in the field's accessibility.

Besides the bug that was called out regarding `firstRowToSearchIndex`, I think the proposed solution works, but it only fixes a particular use case. 

One alternative solution might be to add a options enum `KIFPickerSearchOrder` like:

```
typedef NS_OPTIONS(NSUInteger, KIFPickerSearchOrder) {
    KIFPickerSearchForwardFromStart = 0,
    KIFPickerSearchBackwardFromEnd = 1,
    KIFPickerSearchForwardFromCurrentValue = 2,
    KIFPickerSearchBackwardFromCurrentValue = 3
};
```

I think this would achieve the same results as what you've currently got, except the behavior is less surprising to callers.

My favored approach is to add a new API for `-[KIFUITestActor selectDatePickerDate:(NSDate *)date]` that stores the selected rows for each component, calls through to `-[NSDatePicker setDate:animated:]`, spins the runloop, then calls the delegate for `pickerView:didSelectRow:inComponent:` for each component whose row changed. I think this would be the most straightforward approach. Given that we're already using `selectRow:` picker view API and directly invoking the delegate rather than using flick gestures to scroll the picker view, this feels ok to me. Thoughts?